src/Java.Interop.Export/Java.Interop/MarshalMemberBuilder.cs

```diff
@@ -254,23 +254,22 @@ static Type GetMarshalerType (Type returnType, List<Type> funcTypeParams, Type d
 			funcTypeParams.RemoveRange (0, 2);
 			var marshalDelegateName = new StringBuilder ();
 			marshalDelegateName.Append ("_JniMarshal_PP");
-			foreach (var paramType in funcTypeParams) {
-				marshalDelegateName.Append (GetJniMarshalDelegateParameterIdentifier (paramType));
-			}
-			marshalDelegateName.Append ("_");
-			if (returnType == null) {
-				marshalDelegateName.Append ("V");
-			} else {
-				marshalDelegateName.Append (GetJniMarshalDelegateParameterIdentifier (returnType));
-			}
+			AddMarshalerTypeNameSuffix (marshalDelegateName, returnType, funcTypeParams);
 
 			Type marshalDelegateType = declaringType.Assembly.GetType (marshalDelegateName.ToString (), throwOnError: false);
 
-			// Punt?; System.Linq.Expressions will automagically produce the needed delegate type.
-			// Unfortunately, this won't work with jnimarshalmethod-gen.exe.
 			return marshalDelegateType;
 		}
 
+		public static void AddMarshalerTypeNameSuffix (StringBuilder sb, Type returnType, List<Type> funcTypeParams)
```

Comment:
…[I ask](https://github.com/xamarin/java.interop/pull/711#issuecomment-690732073) because I'm wondering if there's some way to *not* add this `public` export, or at least find a "more suitable" location for it?